src/main/java/com/indeed/jiraactions/api/ApiCaller.java

```diff
@@ -3,178 +3,120 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.indeed.jiraactions.JiraActionsIndexBuilderConfig;
+
+import okhttp3.Cookie;
+import okhttp3.CookieJar;
+import okhttp3.Headers;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
 import org.apache.commons.codec.binary.Base64;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.net.ssl.HttpsURLConnection;
-import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
-import java.util.Objects;
+import java.util.concurrent.TimeUnit;
 
 public class ApiCaller {
     protected final JiraActionsIndexBuilderConfig config;
 
-    private static final Logger log = LoggerFactory.getLogger(ApiCaller.class);
     private static final ObjectMapper objectMapper = new ObjectMapper();
+    private static final CookieJar cookieJar = setupCookieJar();
+    private final OkHttpClient client;
     private final String authentication;
-    private String jsessionId = null;
-    private String upstream = null;
-    private String cookies = "";
-    private String pinnedNode = null;
 
     public ApiCaller(final JiraActionsIndexBuilderConfig config) {
         this.config = config;
         this.authentication = getBasicAuth();
+        this.client = new OkHttpClient.Builder()
+                .connectTimeout(20, TimeUnit.SECONDS)
```

Comment:
Where did these numbers come from?

Reply:
They default to 10 seconds. After messing around with the batch size to around 100 I set it around there because towards the end of indexing all the issues, the time could spike to 15-20 seconds. This shouldn't be a problem now that we are indexing smaller amounts and I can remove it (set it back to default).

Reply:
Feels like the kind of thing that should be sourced from a config file.